resteasy-servlet-initializer/src/main/java/org/jboss/resteasy/plugins/servlet/ResteasyServletInitializer.java

```diff
@@ -67,11 +64,15 @@ else if (clazz.isAnnotationPresent(Provider.class))
       }
       if (appClasses.size() == 0 && resources.size() == 0) return;
 
-      if (appClasses.size() == 0)
-      {
-         // todo make sure we can do this on all servlet containers
-         //handleNoApplicationClass(providers, resources, servletContext);
-         return;
+      if (appClasses.size() > 1) {
+         for (Class<?> app : appClasses) {
+            // App has provided an Application impl.  The local default one
+            // isn't needed.  Remove it.
+            if (app.getSimpleName().equals("DefaultApplicationClazz")) {

```

Comment:
I suggest creating a String constant using something like DefaultApplicationClazz.getClass().getName() and then use that to run the comparison here, so that a future refactoring of the class does not risk breaking this.